raster/src/main/scala/geotrellis/raster/CellFeatures.scala

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2019 Azavea
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package geotrellis.raster
+
+import geotrellis.raster.rasterize.Rasterizer
+import geotrellis.vector.{Feature, Geometry, Point, Polygon, Extent}
+import geotrellis.util.MethodExtensions
+
+import spire.syntax.cfor._
+
+
+/** Type class to convert a raster into features of cell geometries */
+trait CellFeatures[R, D] {
```

Comment:
`R` is unconstrained because I want to be able to do this to `RasterSource` which doesn't have much in common with `Raster[Tile]`